### ISSUE TITLE
Ensure proper handling of division-length and other eccentric texts in suttaplex list

### ISSUE THREAD

**Issue by SamaneriVimala** [suttacentral/suttacentral#789]
Original ticket Creator:
sujato.bhikkhu

Certain texts, especially the so-called "division length texts" don't neatly fall into our normal system of a "division" with "suttas". This includes important texts such as the Pali Dhammapada, and the patimokkhas. It also includes various more obscure texts, such as the Sanskritic texts. Here I will begin by making a list of all problematic texts.* Pali Dhammapada (dhp). Also g2dhp, g3dhp, * Yogacarabhumi (ybs)** This comprises two sections (actually these are just the two relevant portions of a much larger text that we do not include.) The overall text is ybs. The sections are sag and sbh. Currently both the parallels and the sequence are confused. * The patimokkhas are not showing anything except the node headings. This applies to all patimokkhas, so I won't repeat each case.* Other Vinaya texts that don't work: ** [lzh-dg-bu-kammavaca|http://127.0.0.1/lzh-dg-bu-kammavaca]** lzh-dg-bi-kammavaca** lzh-dg-bu-kammavaca-2** sariputrapariprccha** lzh-mi-bu-kammavaca** lzh-mu-bu-kammavaca** lzh-mu-khuddakavatthu** xct-mu-khuddakavatthu** xct-mu-uttaragantha** xct-mu-bu-kammavaca** xct-mu-uttaragantha-2** xct-mu-bi-pm** san-mu-bu-kammavaca** san-mu-mpt-bu-pm** other-matika** sammitiya-sastra** pariprccha** lzh-sarv-bu-kammavaca** lzh-sarv-matika** lzh-sarv-bu-kammavaca-2** san-sarv-bu-kammavaca** san-sarv-sm

h2. How they should work

Let's take the patimokkhas as example, and see how that goes. The patimokkhas are all uniformly formatted in HTML. They have mostly the same structure as the relevant Vibhanga, except the Vibhangas incorporate explanations. Thus we present the patimokkhas all on one page, with textual divisions determined by headings, while the Vibhangas present each rule on a separate page.

Nevertheless, in the sidebar and the suttaplex, they should both be presented the same way, that is:* Rule categories in the sidebar (this is currently correct)* Each rule in a separate suttaplex.

This is because the rules, even though they may be very short, all have meaningful titles and many parallels.
h3. An aside on vaggas
*tl;dr:* don't worry too much about them for the time being. If they work, great, otherwise never mind.

When I said that the patimokkhas have "mostly" the same structure as the Vibhanga, the exception I was thinking about is the vaggas. Like all early texts, the patimokkhas and Vibhangas are divided into vaggas of roughly ten items. This occurs per rule-category, so it only pertains to the categories with many rules, namely nissaggiya, pacittiya, and sekhiya.

However, the application of this is somewhat complex. Even within the Pali tradition, there are two separate ways of dividing the sekhiyas into vaggas. And in the Chinese texts the vaggas are, as always, indicated partially and obscurely, or sometimes not at all. And in the case of the Tibetan texts, we don't have vagga information.

Taking all this into consideration, it may be simplest for now to just omit the vaggas. They are currently not being displayed in the sidebar or suttaplex. If our code decides to pick them up, then great, but otherwise we can leave it for now.

Jira ID: SUCE-452
Jira Creation Date: 13/Jan/18 2:46 AM
Jira Epic: 
Jira Priority: Medium
Jira Status (at import): In Progress

Comments:

**Comment by SamaneriVimala** [comment]
@sujato : Vinaya Patimokkhas are done.

**Comment by SamaneriVimala** [comment]
Other Vinaya texts that don't work:

The following we do not have a text nor a menu structure on the old site.
1. lzh-dg-bu-kammavaca
1. lzh-dg-bi-kammavaca
1. lzh-dg-bu-kammavaca-2
1. sariputrapariprccha
1. lzh-mi-bu-kammavaca

1. lzh-mu-bu-kammavaca
1. lzh-mu-khuddakavatthu
1. xct-mu-khuddakavatthu
1. xct-mu-uttaragantha
1. xct-mu-bu-kammavaca
1. xct-mu-uttaragantha-2
1. san-mu-bu-kammavaca
1. other-matika
1. sammitiya-sastra
1. lzh-sarv-bu-kammavaca
1. lzh-sarv-matika
1. lzh-sarv-bu-kammavaca-2
1. san-sarv-bu-kammavaca
1. san-sarv-bi-kammavaca

The following I can do:
1. xct-mu-bi-pm
1. san-mu-mpt-bu-pm

What is this?:
1. pariprccha

**Comment by SamaneriVimala** [comment]
Pali Dhammapada (dhp). Also g2dhp, g3dhp: fall under ticket #817 

Yogacarabhumi (ybs) - Done partly but waiting for feedback on Dhp ticket before proceeding.

**Comment by SamaneriVimala** [comment]
YBS is also done now.

**Comment by SamaneriVimala** [comment]
YBS is done as well as the rest, except the 20 as indicated above.
So please let me know what to do with those.

**Comment by sujato** [comment]
Okay so as far as the "20" go, it seems that they are either kammavaca (which would mostly or all be ordination procedure) or later treatises. As such they probably don't have much in the way of direct parallels, or if they do, they don't need to be displayed with huge detail. So it would probably be fine to just display each one of these through a single suttaplex.

As for YBS, I think it's okay, the publication details are complex, so just keeping them in the text metadata itself should be fine, unless you really want to add more details in the suttaplex!

**Comment by SamaneriVimala** [comment]
Just to make sure. You want to have a single suttaplex card for each one of these 20 but with nothing in it except the title?

**Comment by sujato** [comment]
… and the text, right?

**Comment by SamaneriVimala** [comment]
We do not have those texts. Where do I find them?

**Comment by sujato** [comment]
Sorry, my bad, I thought we had them. They're all on CBETA, so we can grab them when we like. I remember now, they are included because they are technically EBTs or EBT adjacent, but in fact are of little interest as they are mostly extracts from the main Vinayas. Actually we can leave them as just a suttaplex with title, why not? Some day someone might get around to readying the text!
